### PR TITLE
Add reservation controller

### PR DESCRIPTION
**Pull request by Kweeka1**
## Summary

- Added reservation controller with `create` and `delete` methods

## Changes log

- [x] Updated `routes.rb` to include routes for reservation controller.
- [x] Added `reservations_controller.rb` in `app/controller/api/v1`

## Acceptance criteria
- [x] The controller should have a post method to add a new reservation
- [x] The controller should have a delete method to delete a reservation
- [x] For the post reservation, it should accept `:starting_date`, `:user_id`, `:car_id`, `:ending_date`, and `:city` 
- [x] For the delete method, the user can delete a reservation by its `id`

### CODE REVIEW

**Review by KIsmoilov — CHANGES_REQUESTED** [review]
Hi @Kweeka1  , 

Great work on implementing most of the requirements 👏 
You've done well implementing all the requirements, but I have some suggestions.

almost4

 ## Highlights 💯 
- Good job following the advice of the previous reviewer ✔️ 
- No linter errors. :heavy_check_mark:
- Git flow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️

 ## Required Changes ♻️
- Please see the inline comments
 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Comment by Kweeka1** [comment]
Hey @KIsmoilov, I think you forgot to add inline comments.

**Review by KIsmoilov — CHANGES_REQUESTED** [review]

File: config/routes.rb [hunk 1]
```diff
@@ -9,6 +11,7 @@
   namespace :api do
     namespace :v1 do
       resources :homes, only: [:index]
+      resources :reservations, only: [:create, :destroy]
```
[the remark this answers is not in the document]
Comment (KIsmoilov) [reply config/routes.rb:14]: @Kweeka1 Sorry about that, inline comment was pending on my side and was not added. Please see the above inline comment.

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  get 'reservations/create'
+  get 'reservations/destroy'
```
Comment (wassimchakib) [inline config/routes.rb:5]: [Optional] I think there is no point of having a get method for create and destroy. It should be defined in resources which you already did.

Comment (Kweeka1) [reply config/routes.rb:5]: I believe those lines were auto-generated. I'll proceed to remove them.

File: config/routes.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kweeka1) [reply config/routes.rb:14]: I completely agree with your opinion. Including reservations [:index] method is necessary to display all the reservations. Thank you for bringing this up.

**Review by KIsmoilov — APPROVED** [review]
Hi @Kweeka1 ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :heavy_check_mark:
*   Solid pull request scroll :heavy_check_mark:
*   Good quality commits :heavy_check_mark:
*   All the required changes have been implemented :heavy_check_mark:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**